skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,331 @@
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    src to the dst, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    src : (N, 2)
+        Control point at source coordinates.
+    dst : (N, 2)
+        Control point at target coordinates.
+    output_region : tuple of integers, optional
+        The region ``(xmin, ymin, xmax, ymax)`` of the output
+        image that should be produced. (Note: The region is inclusive, i.e.
+        xmin <= x <= xmax)
+    interpolation_order : int, optional
+        If value is 1, then use linear interpolation else use
+        nearest-neighbor interpolation.
+    grid_scaling : int, optional
+        If grid_scaling is greater than 1, say x, then the transform is
+        defined on a grid x times smaller than the output image region.
+        Then the transform is bilinearly interpolated to the larger region.
+        This is fairly accurate for values up to 10 or so.
+
+    Returns
+    -------
+    warped : array_like
+        The warped input image.
+
+    Raises
+    ------
+    RuntineError
+        If ``x_steps`` or ``y_steps`` <= 0
+
+    References
+    ----------
+    .. [1] Bookstein, Fred L. "Principal warps: Thin-plate splines and the
+           decomposition of deformations," IEEE Transactions on pattern analysis
+           and machine intelligence 11.6 (1989): 567–585.
+
+    Examples
+    --------
+    Produce a warped image rotated by 90 degrees counter-clockwise.
+
+    >>> import skimage as ski
+    >>> astronaut = ski.data.astronaut()
+    >>> image = ski.color.rgb2gray(astronaut)
+    >>> src = np.array([[0, 0], [0, 512], [512, 512],[512, 0]])
+    >>> dst = np.array([[512, 0], [0, 0], [0, 512],[512, 512]])
+    >>> output_region = (0, 0, image.shape[0], image.shape[1])
+    >>> tform = ski.transform.TpsTransform()
+    >>> tform.estimate(src, dst)
+    True
+    >>> warped_image = ski.transform.tps_warp(
+    ...     image, src, dst, output_region=output_region
+    ... )
+
+    """
+    image = np.asarray(image)
+
+    if image.shape[0] == 0 or image.shape[1] == 0:
+        raise ValueError(
+            f"Cannot warp image with invalid shape: {image.shape}"
+        )
+
+    if image.ndim not in (2, 3):
+        raise ValueError("Only 2D and 3D images are supported")
+
+    if output_region is not None:
+        if not isinstance(output_region, tuple) or len(output_region) != 4:
+            raise ValueError("Output region should be a tuple of 4 values.")
+    output_region = output_region or (0, 0, image.shape[0], image.shape[1])
+
+    x_min, y_min, x_max, y_max = output_region
+    if grid_scaling is None:
+        grid_scaling = 1
+    x_steps = (x_max - x_min) // grid_scaling
+    y_steps = (y_max - y_min) // grid_scaling
+
+    if x_steps <= 0 or y_steps <= 0:
+        raise RuntimeError('Unable to compute output region')
```

Comment:
Right now, this message could be more helpful by telling the user how to address the error: modify `grid_scaling` and / or `output_region`. Maybe something like
```python
RuntimeError(f"Invalid or empty `output_region`: {output_region}") 
```